code/EventLinkQR/functions/index.js

```diff
@@ -109,19 +141,48 @@ exports.sendNotificationToEventAttendees = functions.firestore
                 console.log(`Notifications updated for ${uniqueTokens.length} unique devices.`);
 
 
-                // Send updated notifications via FCM
-                const messages = uniqueTokens.map(token => ({
-                    token,
-                    notification: {
-                        title: lastNotification.heading,
-                        body: lastNotification.description,
-                    }
-                }));
+    
+                const eventDoc = await admin.firestore().collection('Events').doc(eventId).get();
+                if (!eventDoc.exists || !eventDoc.data().name) {
+                    console.log(`Event name not found for event: ${eventId}`);
+                }
 
-                const response = await admin.messaging().sendAll(messages);
-                console.log('Successfully sent messages:', response.successCount);
-            } catch (error) {
-                console.error('Error sending updated notifications:', error);
+                const eventName = eventDoc.data().name
+    
+
+                if (uniqueTokens.length > 0) {
+                    const multicastMessage = {
+                        tokens: uniqueTokens,
+                        data: {
+                            // Custom data
+                            title: lastNotification.heading,
+                            body: lastNotification.description,
+                            eventId: eventId,
+                            eventName: eventName,
+                        }
+                    };
+
+                try {
+                    // Correcting the method call to reflect your original query
+                    const response = await admin.messaging().sendEachForMulticast(multicastMessage);
+                    console.log('Successfully sent messages:', response.successCount);
+        
+                    // Handling individual message responses
+                    response.responses.forEach((resp, idx) => {
+                        if (resp.success) {
+                            console.log(`Message sent to token ${uniqueTokens[idx]}`);
+                        } else {
+                            console.error(`Failed to send message to token ${uniqueTokens[idx]}`, resp.error);
+                        }
+                    });
+                } catch (error) {
+                    console.error('Error sending multicast messages:', error);
+                }
+                // const response = await admin.messaging().sendEachForMulticast(messages);
```

Comment:
Is this supposed to be commented out?

Reply:
Sorry, my bad. I have removed those lines now.